homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,186 @@
+"""Platform for Mazda climate integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pymazda import Client as MazdaAPIClient
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.temperature import convert as convert_temperature
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DATA_REGION, DOMAIN
+
+PRESET_DEFROSTER_OFF = "Defroster Off"
+PRESET_DEFROSTER_FRONT = "Front Defroster"
+PRESET_DEFROSTER_REAR = "Rear Defroster"
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+def _front_defroster_enabled(preset_mode: str | None) -> bool:
+    return preset_mode in [
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+        PRESET_DEFROSTER_FRONT,
+    ]
+
+
+def _rear_defroster_enabled(preset_mode: str | None) -> bool:
+    return preset_mode in [
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+        PRESET_DEFROSTER_REAR,
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the climate platform."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    client = entry_data[DATA_CLIENT]
+    coordinator = entry_data[DATA_COORDINATOR]
+    region = entry_data[DATA_REGION]
```

Comment:
This would be a bit cleaner if entry_data was a `@dataclass`

Reply:
I agree. We already discussed this previously here: https://github.com/home-assistant/core/pull/75037#discussion_r956785732. I plan to implement this in a follow-up PR.